src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,53 @@
+package duke.task;
+
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A generic task object.
+ */
+public class Task {
+    protected String desc;
+    protected boolean isDone;
+    protected DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd MMMM yyyy");
+    protected DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HH:mm");
+
+    /**
+     * Creates a new Task object.
+     * @param desc A description of the task.
+     */
+    public Task(String desc) {
+        this.desc = desc;
+        this.isDone = false;
+    }
+    public String getDesc() {
+        return desc;
+    }
+
+    public boolean isDone() {
+        return isDone;
+    }
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```

Comment:
Maybe comments can be placed directly above the code and indented relative to their position in the code?